### PR TITLE
Async theme

### PR DESCRIPTION
**Pull request by FraukeF**
Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/lib/MapView.ts [hunk 1]
```diff
@@ -1727,6 +1641,21 @@ export class MapView extends EventDispatcher {
...
+     */
+    get overlayScene(): THREE.Scene {
+        return this.m_overlayScene;
+    }
+
+    /**
+     * The MapViewEnvironmen used by this `MapView`.
+     * @internal
+     */
+    get sceneEnvironment(): MapViewEnvironment {
```
[the remark this answers is not in the document]
Comment (FraukeF) [reply @here/harp-mapview/lib/MapView.ts:1655]: done

**Comment by FraukeF** [comment]
The removal of references to the deprecated theme setters/getters I ll do in a next PR, same for the support of the FlatTheme

**Review by nzjony — COMMENTED** [review]
Looks good, just some minor comments.

File: @here/harp-mapview/lib/MapView.ts [hunk 2]
```diff
@@ -1445,82 +1436,37 @@ export class MapView extends EventDispatcher {
...
-            dataSource.setTheme(this.m_theme);
-        }
         this.THEME_LOADED_EVENT.time = Date.now();
         this.dispatchEvent(this.THEME_LOADED_EVENT);
         this.update();
+        return newTheme;
+    }
+
+    /**
+     * Returns the currrently set `Theme`as a `Promise` as it might be still loading/updating.
```
[the remark this answers is not in the document]
Comment (FraukeF) [reply @here/harp-mapview/lib/MapView.ts:1466]: done

File: @here/harp-mapview/lib/MapView.ts [hunk 3]
```diff
@@ -2190,8 +2144,12 @@ export class MapView extends EventDispatcher {
                 this.update();
             });
 
+            const theme = await this.getTheme();
+            if (theme !== undefined && theme.styles !== undefined) {
+                dataSource.setTheme(theme);
```
[the remark this answers is not in the document]
Comment (FraukeF) [reply @here/harp-mapview/lib/MapView.ts:2149]: good catch, was just an oversight, but I ll add it to setTheme

[the remark this answers is not in the document]
Comment (FraukeF) [reply @here/harp-mapview/lib/MapView.ts:2151]: the language setting in the setTheme seems broken, Ill keep it as is and investigate why its not working when updating working on the DataSource.setTheme , which is one of the next steps anyway

**Review by nzjony — APPROVED** [review]

**Pull request merged**